packages/editor-ui/src/components/InputPanel.vue

```diff
@@ -169,8 +170,25 @@ export default mixins(
 			const node = this.parentNodes.find((node) => this.currentNode && node.name === this.currentNode.name);
 			return node ? node.depth: -1;
 		},
+		isMultiInputNode (): boolean {
+			if (this.currentNode) {
+				const currentNodeType = this.$store.getters['nodeTypes/getNodeType'](this.currentNode.type, this.currentNode.typeVersion) as INodeTypeDescription;
+				return currentNodeType.inputs.length > 0;
+			}
+			return false;
+		},
 	},
 	methods: {
+		getMultipleNodesText(nodeName?: string):string {
+			if(!this.isMultiInputNode || !this.activeNode) return '';
+
+			const currentNodeConnections = this.currentWorkflow.connectionsByDestinationNode[this.activeNode.name].main || [];
+			const nodeIndex = currentNodeConnections.findIndex( n => n[0] && n[0].node === nodeName);
+
+			return nodeIndex >= 0
+				? this.$locale.baseText('ndv.input.nodePosition', { interpolate: { index: `${nodeIndex + 1}` } })
```

Comment:
Input 1 and Input 2 are actually defined at the node levels.. any node can name these differently.. we should use those names set by the node.. rather than their index
https://github.com/n8n-io/n8n/blob/master/packages/nodes-base/nodes/Merge/Merge.node.ts#L27

Reply:
I've updated it to get the names dynamically